File: ios/XiriGoEcommerce/Core/DesignSystem/Theme/XGTypography.swift
import SwiftUI

/// Typography tokens derived from `shared/design-tokens/foundations/typography.json`.
/// All font styles use Poppins (primary) or Source Sans 3 (price).
/// Values must match the JSON source — never use Font.system().
enum XGTypography {
    // MARK: - Micro

    /// 10pt Regular — delivery deadline text, micro labels
    static let micro = Font.custom("Poppins-Regular", size: 10)

    // MARK: - Caption

    /// 12pt Regular — product card title, review count, badge labels
    static let caption = Font.custom("Poppins-Regular", size: 12)

    /// 12pt Medium — divider text
    static let captionMedium = Font.custom("Poppins-Medium", size: 12)

    /// 12pt SemiBold — product card name on grid cards
    static let captionSemiBold = Font.custom("Poppins-SemiBold", size: 12)

    // MARK: - Body

    /// 14pt Regular — form labels, body text, banner subtitle
    static let body = Font.custom("Poppins-Regular", size: 14)

    /// 14pt Medium — links, filter pill text, "See All"
    static let bodyMedium = Font.custom("Poppins-Medium", size: 14)

    /// 14pt SemiBold — subtitle text, tag text
    static let bodySemiBold = Font.custom("Poppins-SemiBold", size: 14)

    // MARK: - Body Large

    /// 16pt Regular — search placeholder, welcome text
    static let bodyLarge = Font.custom("Poppins-Regular", size: 16)

    // MARK: - Subtitle

    /// 18pt SemiBold — section headings
    static let subtitle = Font.custom("Poppins-SemiBold", size: 18)

    /// 18pt Bold — CTA button text
    static let subtitleBold = Font.custom("Poppins-Bold", size: 18)

    // MARK: - Title

    /// 20pt SemiBold — deal card product name
    static let title = Font.custom("Poppins-SemiBold", size: 20)

    // MARK: - Headline

    /// 24pt SemiBold — hero banner headline
    static let headline = Font.custom("Poppins-SemiBold", size: 24)

    // MARK: - Display

    /// 28pt Bold — large display text, promotional headers
    static let display = Font.custom("Poppins-Bold", size: 28)

    // MARK: - Legacy Aliases (mapped to design token equivalents)

    static let displayLarge = Font.custom("Poppins-Bold", size: 28)
    static let displayMedium = Font.custom("Poppins-Bold", size: 28)
    static let displaySmall = Font.custom("Poppins-Bold", size: 28)
    static let headlineLarge = Font.custom("Poppins-SemiBold", size: 24)
    static let headlineMedium = Font.custom("Poppins-SemiBold", size: 24)
    static let headlineSmall = Font.custom("Poppins-SemiBold", size: 24)
    static let titleLarge = Font.custom("Poppins-SemiBold", size: 20)
    static let titleMedium = Font.custom("Poppins-Medium", size: 16)
    static let titleSmall = Font.custom("Poppins-Medium", size: 14)
    static let bodySmall = Font.custom("Poppins-Regular", size: 12)
    static let labelLarge = Font.custom("Poppins-Medium", size: 14)
    static let labelMedium = Font.custom("Poppins-Medium", size: 12)
    static let labelSmall = Font.custom("Poppins-Medium", size: 11)

    // MARK: - Price Font Helper

    /// Source Sans 3 Black for price display components.
    static func priceFont(size: CGFloat) -> Font {
        .custom("SourceSans3-Black", size: size)
    }

    /// Poppins Medium for strikethrough original prices.
    static func strikethroughFont(size: CGFloat) -> Font {
        .custom("Poppins-Medium", size: size)
    }
}

File: android/app/src/main/java/com/xirigo/ecommerce/core/designsystem/theme/XGTypography.kt
package com.xirigo.ecommerce.core.designsystem.theme

import androidx.compose.material3.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/**
 * Font families from design tokens (typography.json).
 * When actual font files are embedded in res/font/, replace these with:
 *   FontFamily(Font(R.font.poppins_regular, FontWeight.Normal), ...)
 */
val PoppinsFontFamily: FontFamily = FontFamily.Default
val SourceSans3FontFamily: FontFamily = FontFamily.SansSerif

val XGTypography = Typography(
    // display (28sp bold) — from typeScale.display
    displayLarge = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 28.sp,
        lineHeight = 36.sp,
        fontWeight = FontWeight.Bold,
    ),
    displayMedium = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 28.sp,
        lineHeight = 36.sp,
        fontWeight = FontWeight.Bold,
    ),
    displaySmall = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 28.sp,
        lineHeight = 36.sp,
        fontWeight = FontWeight.Bold,
    ),
    // headline (24sp semiBold) — from typeScale.headline
    headlineLarge = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 24.sp,
        lineHeight = 32.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    headlineMedium = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 24.sp,
        lineHeight = 32.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    headlineSmall = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 24.sp,
        lineHeight = 32.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    // title (20sp semiBold) — from typeScale.title
    titleLarge = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 20.sp,
        lineHeight = 28.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    // subtitle (18sp semiBold) — from typeScale.subtitle
    titleMedium = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 18.sp,
        lineHeight = 26.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    // bodySemiBold (14sp semiBold) — from typeScale.bodySemiBold
    titleSmall = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        fontWeight = FontWeight.SemiBold,
    ),
    // bodyLarge (16sp regular) — from typeScale.bodyLarge
    bodyLarge = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 16.sp,
        lineHeight = 22.sp,
        fontWeight = FontWeight.Normal,
    ),
    // body (14sp regular) — from typeScale.body
    bodyMedium = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        fontWeight = FontWeight.Normal,
    ),
    // caption (12sp regular) — from typeScale.caption
    bodySmall = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 12.sp,
        lineHeight = 16.sp,
        fontWeight = FontWeight.Normal,
    ),
    // bodyMedium weight (14sp medium) — from typeScale.bodyMedium
    labelLarge = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 14.sp,
        lineHeight = 20.sp,
        fontWeight = FontWeight.Medium,
    ),
    // captionMedium (12sp medium) — from typeScale.captionMedium
    labelMedium = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 12.sp,
        lineHeight = 16.sp,
        fontWeight = FontWeight.Medium,
    ),
    // micro (10sp regular) — from typeScale.micro
    labelSmall = TextStyle(
        fontFamily = PoppinsFontFamily,
        fontSize = 10.sp,
        lineHeight = 14.sp,
        fontWeight = FontWeight.Normal,
    ),
)
